code/modules/mob/living/simple_animal/bot/construction.dm

```diff
@@ -173,46 +154,24 @@
 						build_step++
 
 		if(7)
-			switch(lasercolor)
-				if("b")
-					if(!istype(W, /obj/item/gun/energy/laser/bluetag))
-						return
-				if("r")
-					if(!istype(W, /obj/item/gun/energy/laser/redtag))
-						return
-				if("")
-					if(!istype(W, /obj/item/gun/energy/disabler))
-						return
-				else
+			if(istype(W, /obj/item/gun/energy/disabler))
+				if(!user.temporarilyRemoveItemFromInventory(W))
 					return
-			if(!user.temporarilyRemoveItemFromInventory(W))
-				return
-			name = "[W.name] ED-209 assembly"
-			to_chat(user, "<span class='notice'>You add [W] to [src].</span>")
-			item_state = "[lasercolor]ed209_taser"
-			icon_state = "[lasercolor]ed209_taser"
-			qdel(W)
-			build_step++
+				name = "[W.name] ED-209 assembly"
+				to_chat(user, "<span class='notice'>You add [W] to [src].</span>")
+				item_state = "ed209_taser"
+				icon_state = "ed209_taser"
+				qdel(W)
+				build_step++
 
 		if(8)
 			if(W.tool_behaviour == TOOL_SCREWDRIVER)
 				to_chat(user, "<span class='notice'>You start attaching the gun to the frame...</span>")
 				if(W.use_tool(src, user, 40, volume=100))
-					name = "armed [name]"
-					to_chat(user, "<span class='notice'>The gun is now securely fastened to the frame.</span>")
-					build_step++
-
-		if(9)
```

Comment:
Even beepsky requires a power cell, I don't think ED209 should be an exception, regardless of what dumb players are using for power cells. 

If you want cell type to matter, make the cooldown between shots affected by the charge of the power cell.

Reply:
I really don't like this idea. Very few other bots need cells, and I don't want to buff these things to take down their cool down. Removing the battery requirement is just the simplest option.

Reply:
I'd argue the better change is to make other bots take cells too, but I wouldn't request that as part of this PR because it's out of scope.

Taking cells away from the bots that do use them is the wrong direction though imo, so I'm against it being done here.

If you don't like the idea of making different qualities of cell matter that's fine, but don't take away the requirement for a power cell.

Reply:
Added it back dc5fa9c